integration_test/cases/repo.exs

```diff
@@ -1043,15 +1043,17 @@ defmodule Ecto.Integration.RepoTest do
       bar_ph = {:placeholder, :bar}
       foo_ph = {:placeholder, :foo}
 
+      intensity_query = from p in Post, where: p.id == ^0, select: p.intensity
+
       entries = [
-        %{intensity: 1.0, title: bar_ph, posted: ~D[2020-12-21], visits: foo_ph},
+        %{intensity: intensity_query, title: bar_ph, posted: ~D[2020-12-21], visits: foo_ph},
```

Comment:
This altered test would fail with the previous code because `intensity` would be the first value acted on in `plan_query_in_rows`. Since the counter in that function used to start at 0, the parameters in this query would not be offset by the number of placeholders.